From bb45ab9854fa0a594fe5deb694000ad4e45497bd Mon Sep 17 00:00:00 2001
From: Rudi <r.ihle@s-t.de>
Date: Thu, 24 Dec 2015 10:01:00 +0100
Subject: [PATCH] video: mxc-hdmi-core: Allow cts/n lookup frequencies to be
 off by one

Due to rounding errors it may happen that the calculated pixel clock does
not exactly match the frequency listed in the lookup table. Modes known
to show this problem are 29 and 30 (1440x576p-50). This patch relaxes the
lookup so that a difference of 1kHz will be tolerated.

Signed-off-by: Rudi <r.ihle@s-t.de>
---
 drivers/mfd/mxc-hdmi-core.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/mfd/mxc-hdmi-core.c b/drivers/mfd/mxc-hdmi-core.c
index 4fbd315..a7be3cd 100644
--- a/drivers/mfd/mxc-hdmi-core.c
+++ b/drivers/mfd/mxc-hdmi-core.c
@@ -441,8 +441,8 @@ static bool hdmi_compute_cts_n(unsigned int freq, unsigned long pixel_clk,
 static void hdmi_lookup_cts_n(unsigned int freq, unsigned long pixel_clk,
 				   unsigned int *n, unsigned int *cts)
 {
-	unsigned int clk = pixel_clk / 1000;
-	unsigned int frq = freq;
+	int clk = pixel_clk / 1000;
+	int frq = freq;
 	int i, j;
 
 	*n = 1;
@@ -468,7 +468,7 @@ static void hdmi_lookup_cts_n(unsigned int freq, unsigned long pixel_clk,
 	}
 
 	for (i = 0; i < ARRAY_SIZE(mxc_hdmi_ctsn_tbl); i++) {
-		if (mxc_hdmi_ctsn_tbl[i].pixclk == clk) {
+		if (abs(mxc_hdmi_ctsn_tbl[i].pixclk - clk) <= 1) {
 			for (j = 0; j < 3; j++) {
 				if (mxc_hdmi_ctsn_tbl[i].ctsn[j].freq == frq) {
 					*n *= mxc_hdmi_ctsn_tbl[i].ctsn[j].n;
-- 
1.9.1

